### PR TITLE
allow to set duration higher than width of shell size

### PR DESCRIPTION
**Pull request by scheleon**
**related:** #6

---

I was not able to set the duration higher than the shell size, so I introduced a new variable factor and now the bar fills accordingly( bar fills from done=0;done <(elapsed/factor)).

### CODE REVIEW

File: progress-bar.sh [hunk 1]
```diff
@@ -1,10 +1,17 @@
-#!/usr/bin/env bash
+#!usr/bin/env bash
```
Comment (edouard-lopez) [inline progress-bar.sh]: This is incorrect, you need a valid path for [shebang](https://stackoverflow.com/a/10383546/802365)

File: progress-bar.sh [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
 
 progress-bar() {
-  local duration=${1}
 
-  already_done() { for ((done=0; done<elapsed; done=done+1)); do printf "▇"; done }
-  remaining() { for ((remain=elapsed; remain<duration; remain=remain+1)); do printf " "; done }
+  local duration=${1}
+  local columns=$(tput cols)
```
Comment (edouard-lopez) [inline progress-bar.sh]: Should be done in two steps (cf. [Shellcheck SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155)):

Could you do the same for `duration` and `spaceAvailable`?

```bash
  local duration
  local columns
  local spaceAvailable

  duration=${1}
  columns=$(tput cols)
  spaceAvailable=$((columns-5))
```

File: progress-bar.sh [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+  local spaceAvailable=$((columns-5))
+ 
+  if (( duration <= spaceAvailable )); then factor=1;
```
Comment (edouard-lopez) [inline progress-bar.sh]:
```suggestion
  if (( duration <= space_available )) then fit_to_screen=1; else fit_to_screen=$(( duration / space_available )); fi

```

File: progress-bar.sh [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+#!usr/bin/env bash
 
 progress-bar() {
-  local duration=${1}
 
-  already_done() { for ((done=0; done<elapsed; done=done+1)); do printf "▇"; done }
-  remaining() { for ((remain=elapsed; remain<duration; remain=remain+1)); do printf " "; done }
+  local duration=${1}
+  local columns=$(tput cols)
+  local spaceAvailable=$((columns-5))
```
Comment (edouard-lopez) [inline progress-bar.sh]: Good naming :+1:

However, as per [Google Style Guide naming convention](https://google.github.io/styleguide/shell.xml#Naming_Conventions), please use `snake_case`: 

    space_available

File: progress-bar.sh [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+ 
+  if (( duration <= spaceAvailable )); then factor=1;
+  else factor=$((duration/spaceAvailable));
```
Comment (edouard-lopez) [inline progress-bar.sh]: Bash only does integer division:

```bash
$ echo $((4/2))
2
$ echo $((4/3))
1
```

**Review by edouard-lopez — APPROVED** [review]
Thanks for the PR

**Review by edouard-lopez — CHANGES_REQUESTED** [review]
I did a review, but I got a new line for each step with your code.
progress-bar

Also, you need to rebase on master as I added some tests

**Comment by shubhamkumar-revefi** [comment]
I made some changes and its working fine, no new line, but still its unable to erase the entire last line when completed 100%.

File: progress-bar.sh [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
 progress-bar() {
-  local duration=${1}
+  local duration
+  local columns
+  local space_available
+  local fit_to_screen  
 
+  duration=${1}
+  columns=$(tput cols)
+  space_available=$(( columns-10 ))
```
Comment (edouard-lopez) [inline progress-bar.sh]: could you replace the magic number `10` by an explicit variable name?

File: progress-bar.sh [hunk 7]
```diff
@@ -1,13 +1,21 @@
...
 
-  already_done() { for ((done=0; done<elapsed; done=done+1)); do printf "▇"; done }
-  remaining() { for ((remain=elapsed; remain<duration; remain=remain+1)); do printf " "; done }
+  if (( duration < space_available )); then fit_to_screen=1; else fit_to_screen=$(( duration / space_available )); fit_to_screen=$(( fit_to_screen + 1 )); fi
```
Comment (edouard-lopez) [inline progress-bar.sh]: better spread on several lines for readability.

**Review by edouard-lopez — CHANGES_REQUESTED** [review]
That look nice!
Could you update based on my review, I will merge next week as I won't be available in the coming days.

**Comment by shubhamkumar-revefi** [comment]
I did a quick review and have updated the file. Thanks for the help and support. I would appreciate any advice or suggestions for the future.

File: progress-bar.sh [hunk 8]
```diff
@@ -3,11 +3,26 @@
...
 
 progress-bar() {
-  local duration=${1}
+  local duration
+  local columns
+  local space_available
+  local fit_to_screen  
+  local space_reserved
 
+  space_reserved=10   # reserved width for the percentage value
```
Comment (edouard-lopez) [inline progress-bar.sh]: `6` characters are enough

**Review by edouard-lopez — APPROVED** [review]

**Pull request merged**